modules/javafx.graphics/src/main/java/javafx/scene/robot/Robot.java

```diff
@@ -0,0 +1,373 @@
+    }
+
+    /**
+     * Returns the current mouse (x,y) screen coordinates as a {@link Point2D}.
+     *
+     * @return the current mouse (x,y) screen coordinates
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public Point2D getMousePosition() {
+        return new Point2D(getMouseX(), getMouseY());
+    }
+
+    /**
+     * Moves the mouse to the specified (x,y) screen coordinates.
+     *
+     * @param x screen coordinate x to move the mouse to
+     * @param y screen coordinate y to move the mouse to
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void mouseMove(double x, double y) {
+        peer.mouseMove(x, y);
+    }
+
+    /**
+     * Moves the mouse to the (x,y) screen coordinates specified by the
+     * given {@code location}.
+     *
+     * @param location the (x,y) coordinates to move the mouse to
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public final void mouseMove(Point2D location) {
+        Objects.requireNonNull(location);
+        mouseMove(location.getX(), location.getY());
+    }
+
+    /**
+     * Presses the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to press
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void mousePress(MouseButton... buttons) {
+        peer.mousePress(buttons);
+    }
+
+    /**
+     * Releases the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to release
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void mouseRelease(MouseButton... buttons) {
+        peer.mouseRelease(buttons);
+    }
+
+    /**
+     * Clicks the specified {@link MouseButton}s.
+     *
+     * @implSpec This is a convenience method that is equivalent to calling
+     * {@link #mousePress(MouseButton...)} followed by {@link #mouseRelease(MouseButton...)}.
+     * @param buttons the mouse buttons to click
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void mouseClick(MouseButton... buttons) {
+        mousePress(buttons);
+        mouseRelease(buttons);
+    }
+
+    /**
+     * Scrolls the mouse wheel by the specified amount. Positive {@code wheelAmt}
+     * scrolls up whereas negative scrolls down.
+     *
+     * @param wheelAmt the (signed) amount to scroll the wheel
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void mouseWheel(int wheelAmt) {
+        peer.mouseWheel(wheelAmt);
```

Comment:
As far as I understand some native api use floats/doubles for such API, may be we can do the same?

Reply:
This is an interesting question. And since it is public API, we should decide now whether to change this. If `wheelAmt` is in pixels (I note that this isn't specified), then double might make more sense. If it is in"clicks" of the mouse wheel or similar, then int seems fine. I haven't looked closely enough to know which is intended. Either way, it needs to be specified.

Reply:
Given that the intention of the API is to simulate mouse-wheel actions for testing, defining this in terms of "clicks" makes the most sense (and matches AWT Robot). I am not sure that we need fractions of a click in that case, so int seems fine. I think we should also define the direction in terms of moving the wheel away from the user (positive) or towards the user (negative), rather than scrolling up / scrolling down.

Separately, I tested this on all three platforms, and I think only Linux is implemented correctly. On Windows it looks like positive and negative are backwards. Mac appears not to be scaling the values correctly (or at all). Also, if we change this to document towards / away from the user, then the Mac implementation will need to take this into account if the "natural scrolling" system property is set, since that will reverse the mapping between scrolling away from / towards the user and scrolling up versus down.

We can file follow-on bugs for all these cases.

Reply:
Shall I change the Javadocs to be more specific now (by defining that it is "clicks" not pixels)?

Reply:
Sure, that would be good. The rest can be done as a follow-up

Reply:
I recall lots of change on macOS in the code which handle wheel events, because magic mouse was able to generate subpixel precision. And the current robot API cannot emulate this behavior. Another issues is how to calculate the click, is it a pixel, a unit in user space, or it is a one line of text? BTW I am fine with current solution, it is just some random comments.

Reply:
Small comments about this part of the spec: "Positive {@code wheelAmt} scrolls up whereas negative scrolls down"

I am not sure that usage of "up" and "down" is correct in all cases. I guess it may be different for win/macos, no?

Reply:
Right, that's one of the reasons I suggested a follow-up issue to address this.